src/components/UserInputBlock.tsx

```diff
@@ -75,7 +75,7 @@ export const UserInputBlock = ({ content, prompt, ack, waitForEvent, fireEvent,
         return
       }
 
-      const persistedState = await persistor.get(contentId)
+      const persistedState = await persistor.get(contentId, '1')
```

Comment:
It should be the page content ID. For `UserInputBlock` you should have both the page ID and the interactive ID. The first is the prefetch key and the latter is the data key.

For the pset problems, you could either try to wrap the persistor in `ProblemSetBlock` that is passed down to children so that child problems (similar to what we do for the `onProblemAttempt` handlers) or (as a hack for the spike) use the React context directly.